api/client.go

```diff
@@ -461,30 +463,41 @@ func NewClient(c *Config) (*Client, error) {
 		if err != nil {
 			return nil, errwrap.Wrapf(fmt.Sprintf("failed to determine tokenSinkPath given the provided agent address %q {{err}}", c.AgentAddress), err)
 		}
+		c.TokenFileSinkPath = tokenSinkPath
```

Comment:
It's usually a bad idea to modify an input as that introduces unexpected side effects to callers.